WordPress/gutenberg
# Add fallback to old gallery edit and save for existing gallery

Related to [refactor of Gallery block to nest image blocks](https://github.com/WordPress/gutenberg/pull/25940).

Because the Gallery refactor is a major braking change, this PR:

- Adds a fallback to the old gallery edit and save functions for existing block content rather than transforming them. 
- Adds an experimental flag so users can opt in to show the refactored Gallery block for new blocks.

### To test

Check out this PR

- Add some markup from existing gallery versions to the editor code view (you can find current and old gallery markup [in the fixtures folder](https://github.com/WordPress/gutenberg/tree/master/packages/e2e-tests/fixtures/blocks)).
- Exit code view and make sure the Gallery has not been transformed into the new InnerBlocks format. Check it can be edited, new images added via media dialog and drag and drop, and displays in front end correctly.
- Add a new Gallery and make sure it is also the existing format.
- Toggle the `Enable the refactored gallery block` experiments flag in `wp-admin/admin.php?page=gutenberg-experiments` and make sure that adding a new gallery adds the refactored gallery block and that it is full functional.
